### ISSUE TITLE
test failure: panic while executing SQL Logic Test nightly

### ISSUE THREAD

**Issue by knz** [cockroachdb/cockroach#22927]
Found in: https://teamcity.cockroachdb.com/viewLog.html?tab=buildLog&buildTypeId=Cockroach_SqlLogicTests&buildId=530925&_focus=8377

```
--- progress: /go/src/github.com/cockroachdb/sqllogictest/test/index/orderby_nosort/100/slt_good_5.test: 731 statements/queries
E180222 07:41:00.836330 6215 sql/conn_executor.go:496  [n1,client=127.0.0.1:60288,user=root] a SQL panic has occurred while executing "SELECT pk, col0 FROM tab1 WHERE (((col0 > 86) AND (col3 <= 2)) OR ((col0 > 0) AND (col1 > 86.38))) ORDER BY 2 DESC, 1": flow: unexpected 10240 leftover bytes
E180222 07:41:00.836534 6215 util/log/crash_reporting.go:181  [n1,client=127.0.0.1:60288,user=root] a panic has occurred!
E180222 07:41:00.837821 6215 util/log/crash_reporting.go:445  [n1,client=127.0.0.1:60288,user=root] Reported as error 11532ed0c9604c1e9f21a0d3970866c8
panic: flow: unexpected 10240 leftover bytes [recovered]
	panic: panic while executing 1 statements: SELECT _, _ FROM _ WHERE (((_ > _) AND (_ <= _)) OR ((_ > _) AND (_ > _))) ORDER BY _ DESC, _; caused by flow: unexpected 10240 leftover bytes

goroutine 6215 [running]:
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).closeWrapper(0xc422c1b000, 0x219b6c0, 0xc42468ee40, 0x1c30d80, 0xc434384fe0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:510 +0x365
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func2(0xc422c1b000, 0x219b6c0, 0xc42468ee40)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:471 +0x61
panic(0x1c30d80, 0xc434384fe0)
	/usr/local/go/src/runtime/panic.go:505 +0x229
```

**Comment by RaduBerinde** [comment]
CC @solongordon

**Comment by andreimatei** [comment]
Solon, it looks like this nightly test has been failing like this since the 13th. https://teamcity.cockroachdb.com/viewType.html?buildTypeId=Cockroach_SqlLogicTests&tab=buildTypeHistoryList&branch_Cockroach_Nightlies=__all_branches__

Do you have any time/interest to look at this?
If you click on the "changes" link next to the first failure, there are some DistSQL changes in there, so maybe they're related... They're about sorting and the failing query does seem to have an ORDER BY.... CC @arjunravinarayan too, maybe he's got some idea.

```
Arjun Narayan (arjunravinarayan)
0 files 
Merge pull request #22609 from arjunravinarayan/sorter_refactor

distsqlrun: refactor sorter to eliminate sorterstrategy.go ···
0 files 
Merge pull request #21703 from arjunravinarayan/sorter_row_source

distsqlrun: refactor sorter to implement RowSource ···
arjun
2 files 
distsqlrun: rename sort strategies to processors

No code change except for renames.

Release note: None ···
2 files 
distsqlrun: move sorterstrategy to sorter

No code changes except for code movement.

Release note: None ···
4 files 
distsqlrun: refactor sorter to implement RowSource
```

**Comment by arjunravinarayan** [comment]
This looks like the memory accounting panic that was supposed to be fixed in #22832. I'll take a look.
